v1/gitlab/required-minimal-approvers.rego

```diff
@@ -0,0 +1,56 @@
+package verify
+
+import future.keywords.in
+
+default allow := false
+default violations := []
+
+
+verify = v {
+    v := {
+        "allow": allow,
+        "violation": {
+            "type": "Verify that the number of approvers is less than the config-number",
+            "details": violations,
+        },
+        "summary": [{
+            "allow": allow,
+            "reason": reason,
+            "violations": count(violations),
+        }],
+    }
+}
+
+allow {
+    count(violations) == 0
+}
+
+reason = v {
+    allow
+    v := "All required number of approvers were reached"
+}
+
+reason = v {
+    not allow
+    v := "At least one required number of approvers was not reached"
+}
+
+
+# Violation has been switched here as a list instead of a set
+violations := [r | 
+
+
+    project := object.remove(input.evidence.predicate.content, {"metadata"})[_]
+        
+    approvers := project.project.result_object.approval_settings.approvers
```

Comment:
Again, better put all such assignments after you pass the check and ensure that you actually need the value.